app/assets/javascripts/admin/addon/components/admin-config-areas/color-palette.gjs

```diff
@@ -0,0 +1,221 @@
+import Component from "@glimmer/component";
+import { cached, tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { LinkTo } from "@ember/routing";
+import { service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import Form from "discourse/components/form";
+import { extractError } from "discourse/lib/ajax-error";
+import { i18n } from "discourse-i18n";
+import AdminConfigAreaCard from "admin/components/admin-config-area-card";
+import ColorPaletteEditor, {
+  LIGHT,
+} from "admin/components/color-palette-editor";
+
+export default class AdminConfigAreasColorPalette extends Component {
+  @service toasts;
+  @service router;
+
+  @tracked editingName = false;
+  @tracked editorMode = LIGHT;
+  @tracked hasUnsavedChanges = false;
+
+  @cached
+  get data() {
+    return {
+      name: this.args.colorPalette.name,
+      user_selectable: this.args.colorPalette.user_selectable,
+      colors: this.args.colorPalette.colors,
+      editingName: this.editingName,
+    };
+  }
+
+  @action
+  toggleEditingName() {
+    this.editingName = !this.editingName;
+  }
+
+  @action
+  onLightColorChange(name, value) {
+    const color = this.data.colors.find((c) => c.name === name);
+    color.hex = value;
+    this.hasUnsavedChanges = true;
+  }
+
+  @action
+  onDarkColorChange(name, value) {
+    const color = this.data.colors.find((c) => c.name === name);
+    color.dark_hex = value;
+    this.hasUnsavedChanges = true;
+  }
+
+  @action
+  async handleSubmit(data) {
+    this.args.colorPalette.name = data.name;
+    this.args.colorPalette.user_selectable = data.user_selectable;
+
+    try {
+      await this.args.colorPalette.save();
+      this.editingName = false;
+      this.hasUnsavedChanges = false;
+      this.toasts.success({
+        data: {
+          message: i18n("saved"),
+        },
+      });
+    } catch (error) {
+      this.toasts.error({
+        duration: 3000,
+        data: {
+          message: extractError(error),
+        },
+      });
+    }
+  }
+
+  @action
+  onEditorTabSwitch(newMode) {
+    this.editorMode = newMode;
+  }
+
+  @action
+  async duplicate() {
+    const copy = this.args.colorPalette.copy();
+    copy.name = i18n("admin.config_areas.color_palettes.copy_of", {
+      name: this.args.colorPalette.name,
+    });
+    await copy.save();
+    this.router.replaceWith("adminConfig.color-palettes-show", copy);
```

Comment:
Do you think we need some toasts? I played with it today and it happens very quick and almost didn't notice for the first time that I am already in duplicate :)

Reply:
Yup, makes sense. I've added a toast message.